presto-hive/src/main/java/com/facebook/presto/hive/HiveRecordSink.java

```diff
@@ -319,4 +335,38 @@ public static boolean isTypeSupported(Type type)
             return false;
         }
     }
+
+    private Object translateDateTime(Type type, Object value)
+    {
+        if (value == null) {
+            return null;
+        }
+        if (isArrayType(type)) {
+            List<Object> newValue = new ArrayList<>();
+            Type elementType = typeManager.getType(type.getTypeParameters().get(0).getTypeSignature());
```

Comment:
This can be simplified to `type.getTypeParameters().get(0)` then you don't need to inject a TypeManager